soccer/planning/IndependentMultiRobotPathPlanner.cpp

```diff
@@ -33,6 +35,14 @@ std::map<int, std::unique_ptr<Path>> IndependentMultiRobotPathPlanner::run(
         staticRobotObstacles[shell] = std::make_shared<Geometry2d::Circle>(
             request.start.pos, Robot_Radius);
     }
+
+    auto comparator = [&](int &lhs, int &rhs) {
+            return sortByPriority(requests[lhs], requests[rhs]);
+        };
+
+    std::sort(std::begin(staticRequests), std::end(staticRequests), comparator);
+    std::sort(std::begin(dynamicRequests), std::end(dynamicRequests), comparator);
+
     std::vector<int> inOrderRequests;
```

Comment:
Well, the problem is that the things planned first don't consider the other robots as obstacles since we expect the later robots to plan around us. Its a edge case, but if we planned the dynamics first, the dynamics would run into the static robots. The dynamics would ignore the static since its planned later and the statics can't check where robots will be in the future.